aws-xray-recorder-sdk-core/src/main/java/com/amazonaws/xray/contexts/LambdaSegmentContext.java

```diff
@@ -81,14 +92,19 @@ public Subsegment beginSubsegment(AWSXRayRecorder recorder, String name) {
             // Ensure customers have not leaked subsegments across invocations
             TraceID environmentRootTraceId = LambdaSegmentContext.getTraceHeaderFromEnvironment().getRootTraceId();
             if (environmentRootTraceId != null &&
-                !environmentRootTraceId.equals(parentSubsegment.getParentSegment().getTraceId())) {
+                    !environmentRootTraceId.equals(parentSubsegment.getParentSegment().getTraceId())) {
                 clearTraceEntity();
                 return beginSubsegment(recorder, name);
             }
             Segment parentSegment = parentSubsegment.getParentSegment();
-            Subsegment subsegment = parentSegment.isRecording()
-                                    ? new SubsegmentImpl(recorder, name, parentSegment)
-                                    : Subsegment.noOp(parentSegment, recorder);
+
+            boolean isRecording = (parentSegment.isRecording() &&
+                    samplingStrategyOverride == SamplingStrategyOverride.DISABLED) ||
+                    samplingStrategyOverride == SamplingStrategyOverride.TRUE;
```

Comment:
When parentSegment.isRecording() == false and samplingStrategyOverride == SamplingStrategyOverride.TRUE, should we create a new subsegment?

Reply:
Yes. The override to true will ignore any other indicators. 
I guess this creates a problem of an orphaned subsegment?